src/internal.c

```diff
@@ -1560,8 +1563,6 @@ void SSL_CtxResourceFree(WOLFSSL_CTX* ctx)
     #ifdef OPENSSL_EXTRA
         while (ctx->ca_names != NULL) {
             WOLFSSL_STACK *next = ctx->ca_names->next;
-            wolfSSL_X509_NAME_free(ctx->ca_names->data.name);
```

Comment:
Why are these two lines removed? This is a WOLFSSL_X509_NAME type and it looks like wolfSSL_X509_NAME_free should still be called. I ran valgrind and did not see a leak, but this code was added because of `wolfSSL_CTX_set_client_CA_list`, which does not have a unit test. Would you mind adding a unit test for `wolfSSL_CTX_set_client_CA_list` and `wolfSSL_SSL_CTX_get_client_CA_list`.

Reply:
Thank you for good suggestion !
When I called wolfSSL_X509_NAME_free, I got a double free error, so I held this issue and forgot it.
I fixed memory corruption bug in CTX_free.
I added test for wolfSSL_CTX_set_client_CA_list and wolfSSL_CTX_get_client_CA_list.
And I added Chris update in wc_port.h too.

Reply:
I found a bug in wolfSSL_CTX_add_client_CA at 11087.
ctx->ca_names->data.name  = node->data.name;
Correct code is 
ctx->ca_names->data.name  = &509->subject;
But if  I added wolfSSL_X509_NAME_free in SSL_CtxResourceFree, I got memory leak error in valgrind.
If I removed wolfSSL_X509_NAME_free, memory leak was gone.
I don't know why and still continue to investigate.
If you have any idea, please let me know.

Reply:
Hi Go,

You'll need to determine who owns the `WOLFSSL_X509_NAME` and who is responsible for freeing it. You don't have a memory leak, but you do have access to invalid memory. This could be due to casting the pointer to an incompatible type or using a pointer after its been freed.

It looks to me like you've altered the behavior of `ctx->ca_names` and you'll need to also fix `wolfSSL_CTX_add_client_CA` and any other place that uses `ctx->ca_names`.

Here is the current Jenkins valgrind report:

```
==16242== Invalid read of size 8
==16242==    at 0x46C050: FreeX509Name (in /var/lib/jenkins/workspace/PRB-valgrind-check-v-1-0/tests/unit.test)
==16242==    by 0x46B1EC: SSL_CtxResourceFree (in /var/lib/jenkins/workspace/PRB-valgrind-check-v-1-0/tests/unit.test)
==16242==    by 0x46B26C: FreeSSL_Ctx (in /var/lib/jenkins/workspace/PRB-valgrind-check-v-1-0/tests/unit.test)
==16242==    by 0x40A065: test_wolfSSL_CTX_add_client_CA (in /var/lib/jenkins/workspace/PRB-valgrind-check-v-1-0/tests/unit.test)
```

Reply:
I just copy pointer of WOLFSSL_X509_NAME in wolfSSL_CTX_add_client_CA().
So WOLFSSL_X509_NAME data should be freed in wolfSSL_X509_free,I guess.
Maybe I access invalid pointer of wolfSSL_X509_NAME after wolfSSL_X509_free frees it.
It seems I should not calll wolfSSL_X509_NAME_free() in SSL_CtxResourceFree().
What do you think ?

Reply:
It sounds like you are making a copy a pointer who is owned by another object. The easiest solution would be to duplicate the X509_NAME instead of copying the pointer. Then you can keep the free and you shouldn't have any issues.

Reply:
I changed wolfSSL_CTX_add_client_CA() along your suggestion.
And I also do SSL_CtxResourceFree() to free ca_names data in internal.c.
It passed all tests and looks working.
Could you check it ?

Reply:
Yes, will do tomorrow. Thanks!